graph/db/graph.go

```diff
@@ -3907,6 +3897,64 @@ func (c *ChannelGraph) IsClosedScid(scid lnwire.ShortChannelID) (bool, error) {
 	return isClosed, nil
 }
 
+// GraphSession will provide the call-back with access to a graphdb.CachedGraph
+// instance which can be used to perform queries against the channel graph.
+// If the graph cache is not enabled, then the call-back will be provided with
+// access to the graph via a consistent read-only transaction.
+func (c *ChannelGraph) GraphSession(cb func(graph CachedGraph) error) error {
+	var (
+		tx     kvdb.RTx
+		err    error
+		commit = func() {}
+	)
+	if c.graphCache == nil {
+		tx, err = c.db.BeginReadTx()
```

Comment:
We could instead use a `c.db.View()` and execute the passed in callback within the view closure. This way we could completely remove `BeginReadTx` etc from kvdb.

Reply:
great idea!!

Reply:
ah, the only tricky part here though is that then it wont be a pure refactor right since currently we just log if `BeginReadTx` fails or if `Rollback` fails whereas if we use View then we will fully return errors there. 

so perhaps another follow up just to keep this a pure refactor. 

thoughts?

Reply:
addressed here: https://github.com/lightningnetwork/lnd/pull/9523